rclcpp/include/rclcpp/node.hpp

```diff
@@ -153,6 +153,11 @@ class Node : public std::enable_shared_from_this<Node>
   const std::vector<rclcpp::CallbackGroup::WeakPtr> &
   get_callback_groups() const;
 
+  /// Iterate over the callback groups in the node, calling func on each valid one.
+  RCLCPP_PUBLIC
+  void
+  callback_groups_iterate(const node_interfaces::NodeBaseInterface::CallbackGroupCallback & func);
```

Comment:
+1 to `callback_groups_for_each()`, iterate() sounds like it will return an iterator.
Alternative: `for_each_callback_group()`, it is easier to read IMO.

Reply:
LOL.  I totally meant this to be `callback_groups_for_each`; I just forgot to commit that change :).  Will update.

Reply:
Renamed now :).